packages/components/src/internal/components/search/actions.ts

```diff
@@ -145,3 +159,119 @@ export function getProcessedSearchHits(
+            Query.deleteQueryView({
+                schemaName: model.schemaQuery.schemaName,
+                queryName: model.schemaQuery.queryName,
+                viewName: SAMPLE_FINDER_VIEW_NAME,
+                revert: true,
+                success: () => {
+                    resolve(true);
+                },
+                failure: (error) => {
+                    console.error("There was a problem deleting the Sample Finder view.", error);
+                    reject(resolveErrorMessage(error));
+                }
+            });
+        }
+    });
+}
+
+export function saveFinderGridView(schemaQuery: SchemaQuery, columns: any): Promise<SchemaQuery> {
+    return new Promise((resolve, reject) => {
+        Query.saveQueryViews({
+            schemaName: schemaQuery.schemaName,
+            queryName: schemaQuery.queryName,
+            // Mark the view as hidden, so it doesn't show up in LKS and in the grid view menus
+            views: [{ name: SAMPLE_FINDER_VIEW_NAME, columns, hidden: true }],
+            success: () => {
+                resolve(schemaQuery);
+            },
+            failure: response => {
+                console.error(response);
+                reject(
+                    'There was a problem creating the view for the data grid. ' + resolveErrorMessage(response)
+                );
+            },
+        });
+    });
+}
+
+function getSampleFinderConfigId(finderId: string, suffix: string): string {
+    return finderId + "|" + suffix;
+}
+
+export function getSampleFinderQueryConfigs(user: User, cards: FilterProps[], finderId: string): Promise<{[key: string]: QueryConfig}> {
+    const omittedColumns = getOmittedSampleTypeColumns(user);
+    const baseFilters = [];
+    const requiredColumns = [...SAMPLE_STATUS_REQUIRED_COLUMNS];
+    cards.forEach(card => {
+        if (card.filterArray.length) {
+            card.filterArray.forEach(filter => {
+                requiredColumns.push(filter.getColumnName());
+            });
+            baseFilters.push(...card.filterArray);
+        } else {
+            const cardColumnName = getFilterCardColumnName(card.entityDataType, card.schemaQuery);
+
+            requiredColumns.push(cardColumnName);
+            baseFilters.push(Filter.create(cardColumnName + "/Name", null, Filter.Types.NONBLANK));
+        }
+    });
+    const allSamplesKey = getSampleFinderConfigId(finderId,'exp/materials');
+    const configs: { [key: string]: QueryConfig } = {
+        [allSamplesKey]: {
+            id: allSamplesKey,
+            title: 'All Samples',
+            schemaQuery: SchemaQuery.create(SCHEMAS.EXP_TABLES.MATERIALS.schemaName, SCHEMAS.EXP_TABLES.MATERIALS.queryName, SAMPLE_FINDER_VIEW_NAME),
+            requiredColumns,
+            omittedColumns: ['Run'],
+            baseFilters,
+        },
+    };
+    return new Promise(async (resolve, reject) => {
+        try {
+            const names = await getFinderSampleTypeNames(getContainerFilter());
```

Comment:
as mentioned else where, it seems a little odd to have to make a query for the set of Sample Type names each time we generate the query configs. Could that be done once and passed in here?

Reply:
that might also make this easier to jest test